package/libs/openssl/Makefile

```diff
@@ -156,6 +156,8 @@ else
     OPENSSL_TARGET:=linux-mips-openwrt
 #  else ifeq ($(CONFIG_arm)$(CONFIG_armeb),y)
 #    OPENSSL_TARGET:=linux-armv4-openwrt
+  else ifeq ($(BOARD),ipq806x)
+    OPENSSL_TARGET:=linux-armv4-openwrt
```

Comment:
Can you make this a Kconfig option in Config.in that depends on arm || armeb and defaults to y for TARGET_ipq806x ? That way it's easier to test for others (and configurable).

Also please add `PKG_FLAGS:=nonshared` to the Makefile.